### ISSUE TITLE
cmd.script broken for windows clients

### ISSUE THREAD

**Issue by damntoken** [saltstack/salt#47573]
### Description of Issue/Question
cmd.script does not work with winodws minions.. same error in statefile and direct command.
@ https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.cmdmod.html
Manual has windows example, so I guess it did work for someone at some time..
`salt '*' cmd.script salt://scripts/windows_task.ps1 args=' -Input c:\tmp\infile.txt' shell='powershell'`

### Steps to Reproduce Issue
Enable filesystem.. create testscrip with content "echo 999".
/srv/salt
```
└#> cat testscrip
echo 999
```

```
└#> salt 'linuxminion' cmd.script "salt://testscrip"
linuxminion:
    ----------
    pid:
        126960
    retcode:
        0
    stderr:
    stdout:
        999 <--- works as expected
```

```
└#> salt 'winminion' cmd.script "salt://testscrip"
winminion:
    ----------
    pid:
        3580
    retcode:
        1
    stderr:
        The filename, directory name, or volume label syntax is incorrect.
    stdout:
```

```
└#> salt 'winminion' cmd.script "salt://testscrip" shell='powershell'
winminion:
    ----------
    pid:
        5736
    retcode:
        -196608
    stderr:
        Processing -File ''c:\windows\temp\__salt.tmp.b5wxl5'' failed: The given path's format is not supported. Specify a valid path for the -File parameter.
    stdout:
        Windows PowerShell
        Copyright (C) 2016 Microsoft Corporation. All rights reserved.
```

### Versions Report
Minions all same v with master.. (linuxminion Ubuntu 16.04, Windows is 2016 server, one is fully updated other is not, same symptoms for both. )

@ master
```
└#> salt --versions-report
Salt Version:
           Salt: 2018.3.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, -)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-119-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by Ch3LL** [comment]
should be fixed by https://github.com/saltstack/salt/pull/45716  which will be included in 2018.3.1.

Can you give that fix a try?

**Comment by damntoken** [comment]
My server doesn't use git for certain reasons.
I could do manual few file diff fix but the link provides fix for 2017 not 2018.3.0.
So I guess I have to wait for the release of 2018.3.1 into repositories.
https://repo.saltstack.com/#ubuntu
Thanks though - good to know it's already fixed in upcoming upgrade.
